### PR TITLE
chore: removed eslint-disable statements

### PR DESCRIPTION
**Pull request by sohailfatima**
### Description

Removed all unnecessary / unintentional eslint-disable statements, refactoring the code accordingly. No changes have been made to the functionality.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @sohailfatima! Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.


:warning: We can't start reviewing your pull request until you've submitted a     [signed contributor agreement](https://openedx.org/cla)     or indicated your institutional affiliation. Please see the [CONTRIBUTING](https://github.com/openedx/.github/blob/master/CONTRIBUTING.md)     file for more information.     If you've signed an agreement in the past, you may need to re-sign.  See     [The New Home of the Open edX Codebase](https://open.edx.org/blog/the-new-home-of-the-open-edx-codebase/)     for details.

    
Once you've signed the CLA, please allow 1 business day for it to be processed.  After this time, you can re-run the     CLA check by adding a comment here that you have signed it.  If the problem persists, you can tag the `@openedx/cla-problems` team in a     comment on your PR for further assistance.

**Comment by mphilbrick211** [comment]
Thank you for this contribution, @sohailfatima! Please let me know if you have any questions regarding submitting the [CLA form](https://openedx.atlassian.net/wiki/spaces/COMM/pages/941457737/How+to+start+contributing+to+the+Open+edX+code+base). Thanks!

File: src/components/FilterBar.jsx [hunk 1]
```diff
@@ -194,8 +193,8 @@ const FilterBar = ({
 
 FilterBar.propTypes = {
   intl: intlShape.isRequired,
-  filters: PropTypes.array.isRequired,
-  selectedFilters: PropTypes.object.isRequired,
+  filters: PropTypes.arrayOf(PropTypes.string).isRequired,
```
Comment (awais-ansari) [inline src/components/FilterBar.jsx:196]: it's an array of objects.

File: src/components/NavigationBar/index.js [hunk 2]
```diff
@@ -1,2 +1,3 @@
-/* eslint-disable import/prefer-default-export */
-export { default as CourseTabsNavigation } from './CourseTabsNavigation';
+import CourseTabsNavigation from './CourseTabsNavigation';
+
+export default CourseTabsNavigation;
```
Comment (awais-ansari) [inline src/components/NavigationBar/index.js:3]: this is a single export that can be exported from an original file.

File: src/data/__factories__/index.js [hunk 3]
```diff
@@ -1 +1,3 @@
-export * from './blocks';
+import getBlocksAPIResponse from './blocks';
+
+export default getBlocksAPIResponse;
```
Comment (awais-ansari) [inline src/data/__factories__/index.js:3]: directly import function from the original file. No need to create an index file for just export one file.

File: src/data/slices.js [hunk 4]
```diff
@@ -1,4 +1,4 @@
-/* eslint-disable no-param-reassign,import/prefer-default-export */
+/* eslint-disable no-param-reassign */
```
Comment (awais-ansari) [inline src/data/slices.js:1]: remove this disable eslint and update the code.

File: src/discussions/learners/learner/proptypes.js [hunk 5]
```diff
@@ -1,11 +1,12 @@
-/* eslint-disable import/prefer-default-export */
 import PropTypes from 'prop-types';
 
-export const learnerShape = PropTypes.shape({
+const learnerShape = PropTypes.shape({
```
Comment (awais-ansari) [inline src/discussions/learners/learner/proptypes.js:3]: Delete unused proptypes.

File: src/discussions/post-comments/comments/comment/proptypes.js [hunk 6]
```diff
@@ -1,7 +1,6 @@
-/* eslint-disable import/prefer-default-export */
 import PropTypes from 'prop-types';
 
-export const commentShape = PropTypes.shape({
+const commentShape = PropTypes.shape({
```
Comment (awais-ansari) [inline src/discussions/post-comments/comments/comment/proptypes.js:3]: Remove unused proptypes.

**Review by awais-ansari — CHANGES_REQUESTED** [review]
1. Remove all the icon components and use these icons from paragon. 
2. Remove `eslint-disable no-param-reassig` disable eslint and update the code. 
3. Remove all the index files that  are just using for exporting one component.

File: src/discussions/learners/data/slices.js [hunk 7]
```diff
@@ -28,38 +27,73 @@ const learnersSlice = createSlice({
...
+    //   state.status = RequestStatus.SUCCESSFUL;
+    //   state.pages[payload.page - 1] = payload.results;
+    //   state.learnerProfiles = {
+    //     ...state.learnerProfiles,
+    //     ...(payload.learnerProfiles || {}),
+    //   };
+    //   state.nextPage = (payload.page < payload.pagination.numPages) ? payload.page + 1 : null;
+    //   state.totalPages = payload.pagination.numPages;
+    //   state.totalLearners = payload.pagination.count;
+    // },
```
Comment (awais-ansari) [inline src/discussions/learners/data/slices.js:40]: please remove unnecessary or commented code.

**Review by awais-ansari — CHANGES_REQUESTED** [review]
Please test Discussions  MFE properly once again because this change is big. Other than this PR LGTM.

**Review by awais-ansari — APPROVED** [review]

File: src/discussions/learners/LearnersView.jsx [hunk 8]
```diff
@@ -60,8 +60,7 @@ const LearnersView = () => {
       courseConfigLoadingStatus === RequestStatus.SUCCESSFUL && learners.map((learner) => (
         <LearnerCard learner={learner} key={learner.username} />
       ))
-    // eslint-disable-next-line react/jsx-no-useless-fragment
-    ) || <></>
+    ) || null
```
Comment (sundasnoreen12) [inline src/discussions/learners/LearnersView.jsx:63]: update this function 
const renderLearnersList = useMemo(() => {
  if (courseConfigLoadingStatus === RequestStatus.SUCCESSFUL) {
    return learners.map((learner) => <LearnerCard learner={learner} key={learner.username} />);
  }
  return null;
}, [courseConfigLoadingStatus, learners]);

File: src/discussions/post-comments/postCommentsContext.js [hunk 9]
```diff
@@ -1,8 +1,9 @@
...
 import React from 'react';
 
-export const PostCommentsContext = React.createContext({
+const PostCommentsContext = React.createContext({
   isClosed: false,
   postType: 'discussion',
   postId: '',
 });
+
+export default PostCommentsContext;
```
Comment (sundasnoreen12) [inline src/discussions/post-comments/postCommentsContext.js:9]: add an empty line after export

File: src/discussions/posts/data/hooks.js [hunk 10]
```diff
@@ -24,3 +23,5 @@ export const usePostList = (ids) => {
 
   return sortedIds;
 };
+
+export default usePostList;
```
Comment (sundasnoreen12) [inline src/discussions/posts/data/hooks.js:27]: add empty line after export

File: src/discussions/posts/post/PostFooter.jsx [hunk 11]
```diff
@@ -7,9 +7,10 @@ import { useIntl } from '@edx/frontend-platform/i18n';
 import {
   Icon, IconButton, OverlayTrigger, Tooltip,
 } from '@edx/paragon';
-import { Locked, People } from '@edx/paragon/icons';
+import {
+  Locked, People, StarFilled, StarOutline,
+} from '@edx/paragon/icons';
```
Comment (sundasnoreen12) [inline src/discussions/posts/post/PostFooter.jsx:12]: move these into single line

File: src/discussions/tours/data/selectors.js [hunk 12]
```diff
@@ -1,2 +1,3 @@
-// eslint-disable-next-line import/prefer-default-export
-export const selectTours = (state) => state.tours.tours;
+const selectTours = (state) => state.tours.tours;
+
+export default selectTours;
```
Comment (sundasnoreen12) [inline src/discussions/tours/data/selectors.js:3]: add empty line after export

File: src/test-utils.js [hunk 13]
```diff
@@ -1,6 +1,6 @@
-/* eslint-disable import/prefer-default-export */
-
-export const executeThunk = async (thunk, dispatch, getState) => {
+const executeThunk = async (thunk, dispatch, getState) => {
   await thunk(dispatch, getState);
   await new Promise(setImmediate);
 };
+
+export default executeThunk;
```
Comment (sundasnoreen12) [inline src/test-utils.js:6]: add new line after export statement

**Review by sundasnoreen12 — CHANGES_REQUESTED** [review]
nit: few minor changes

File: src/test-utils.js (hunk 13 above)
Comment (sohailfatima) [reply src/test-utils.js:6]: All the files have a final newline already, it just doesn't show in the changes view.

File: src/discussions/posts/post/PostFooter.jsx (hunk 11 above)
Comment (sohailfatima) [reply src/discussions/posts/post/PostFooter.jsx:12]: Throws eslint `object-curly-newline` error

**Review by sundasnoreen12 — APPROVED** [review]
PR LGTM. Please rebase with master

**Comment by awais-ansari** [comment]
I'm closing this PR as @sohailfatima does not have the cla permission. We created a new PR for these Changes. [658](https://github.com/openedx/frontend-app-discussions/pull/658)

**Comment by openedx-webhooks** [comment]
@sohailfatima Even though your pull request wasn’t merged, please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/frontend-app-discussions&entry.867055334=https://github.com/openedx/frontend-app-discussions/pull/646&entry.1484655318=https://github.com/sohailfatima&entry.752974735=2024-01-19+10:52&entry.1917517419=2024-01-31+11:29&entry.2133058324=No) so we can improve your experience in the future.

**Pull request closed without merging**